pkg/policy/l4.go

```diff
@@ -189,10 +189,22 @@ func CreateL4Filter(peerEndpoints api.EndpointSelectorSlice, rule api.PortRule,
 // specified endpoints and port/protocol for ingress traffic, with reference
 // to the original rules that the filter is derived from. This filter may be
 // associated with a series of L7 rules via the `rule` parameter.
-func CreateL4IngressFilter(fromEndpoints api.EndpointSelectorSlice, rule api.PortRule, port api.PortProtocol,
+func CreateL4IngressFilter(fromEndpoints api.EndpointSelectorSlice, endpointsWithL3Override []labels.LabelArray, rule api.PortRule, port api.PortProtocol,
 	protocol api.L4Proto, ruleLabels labels.LabelArray) L4Filter {
 
-	return CreateL4Filter(fromEndpoints, rule, port, protocol, ruleLabels, true)
+	filter := CreateL4Filter(fromEndpoints, rule, port, protocol, ruleLabels, true)
+
+	// If the filter would apply L7 restrictions for
+	// endpointsWithL3Override (eg, host / world in the relevant daemon
+	// modes), then wildcard those specific endpoints at L7.
+	if rule.Rules != nil && rule.Rules.Len() > 0 {
+		for _, labels := range endpointsWithL3Override {
+			selector := api.ReservedEndpointSelectors[labels[0].Key]
```

Comment:
Is `labels[0]` guaranteed to exist? Wondering in the case of a potential deference of something that does not exist.

Reply:
Yes, it's created two functions up. I played around with a few options, maybe passing a `map[string]EndpointSelector` down would be more readable.

Reply:
Turns out that we don't need this hack any more, we can just pass a `[]api.EndpointSelector` around. I missed this when refactoring to simplify. I'll fix this up.